### PR TITLE
KAA-790: Upgrade Botan version to 1.11

### PR DESCRIPTION
**Pull request by RostakaGmfun**

### CODE REVIEW

File: client/client-multi/client-cpp/impl/security/KeyUtils.cpp [hunk 1]
```diff
@@ -35,8 +40,9 @@ SessionKey KeyUtils::generateSessionKey(std::size_t length)
 bool KeyUtils::checkKeyPair(const KeyPair &keys)
 {
     try {
-        auto &pub = keys.getPublicKey();
-        std::unique_ptr<Botan::Public_Key> ppub{Botan::X509::load_key(pub)};
+        const PublicKey& pub = keys.getPublicKey();
+        Botan::DataSource_Memory dataSrc(pub);
+        std::unique_ptr<Botan::Public_Key> ppub{Botan::X509::load_key(dataSrc)};
```
Comment (ghost) [inline client/client-multi/client-cpp/impl/security/KeyUtils.cpp:45]: Check if this compiles on MSVC 2013

File: client/client-multi/client-cpp/impl/security/RsaEncoderDecoder.cpp [hunk 2]
```diff
@@ -51,13 +52,14 @@ RsaEncoderDecoder::RsaEncoderDecoder(const PublicKey& pubKey,
...
-            remoteKey_->x509_subject_public_key().begin(), remoteKey_->x509_subject_public_key().size()) : "empty"));
+            remoteKey_->x509_subject_public_key().data(), remoteKey_->x509_subject_public_key().size()) : "empty"));
 }
 
-Botan::SecureVector<std::uint8_t> RsaEncoderDecoder::getEncodedSessionKey()
+EncodedSessionKey RsaEncoderDecoder::getEncodedSessionKey()
 {
     Botan::PK_Encryptor_EME enc(*remoteKey_, "EME-PKCS1-v1_5");
-    return enc.encrypt(sessionKey_.bits_of(), rng_);
+    const auto &v = enc.encrypt(sessionKey_.bits_of(), rng_);
```
Comment (ghost) [inline client/client-multi/client-cpp/impl/security/RsaEncoderDecoder.cpp]: Does encrypt() return const reference?

File: client/client-multi/client-cpp/impl/security/RsaEncoderDecoder.cpp [hunk 3]
```diff
@@ -79,10 +81,11 @@ std::string RsaEncoderDecoder::decodeData(const std::uint8_t *data, std::size_t
     return cipherPipe(data, size, Botan::DECRYPTION);
 }
 
-Botan::SecureVector<std::uint8_t> RsaEncoderDecoder::signData(const std::uint8_t *data, std::size_t size)
+Signature RsaEncoderDecoder::signData(const std::uint8_t *data, std::size_t size)
 {
     Botan::PK_Signer signer(*privKey_, "EMSA3(SHA-1)");
-    return signer.sign_message(data, size, rng_);
+    const auto &sgn = signer.sign_message(data, size, rng_);
```
Comment (ghost) [inline client/client-multi/client-cpp/impl/security/RsaEncoderDecoder.cpp]: Does sign_message() return const reference?

File: client/client-multi/client-cpp/impl/security/RsaEncoderDecoder.cpp (hunk 2 above)
Comment (RostakaGmfun) [inline client/client-multi/client-cpp/impl/security/RsaEncoderDecoder.cpp]: Non-const references can't be initialized from rvalues. However, in C++11 it is possible to assign rvalue to non-const reference with move semantics (T&&). - going to change this.

File: client/client-multi/client-cpp/impl/security/RsaEncoderDecoder.cpp (hunk 3 above)
Comment (RostakaGmfun) [inline client/client-multi/client-cpp/impl/security/RsaEncoderDecoder.cpp]: Same as the previous case.

File: client/client-multi/client-cpp/impl/security/KeyUtils.cpp (hunk 1 above)
Comment (RostakaGmfun) [inline client/client-multi/client-cpp/impl/security/KeyUtils.cpp:45]: There are no significant changes and the above code compiled well with MSVC 2013 before these changes.

**Comment by ghost** [comment]
LGTM

**Pull request merged**